check_limits.py

```diff
@@ -1,18 +1,33 @@
+import sys  
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+no_para = 3
 
-  return True
+Battery_range= [[0,45], [20,80], [-sys.maxsize,0.8]]
+Battery_parameters =['temperature', 'soc', 'charge_rate'] 
+
+def breach_L_H(Battery_readings):
+  Reading_status=['n','n','n']
+  for i in range(0, no_para):
+    print(Battery_range[i], Battery_readings[i])
+    if (Battery_readings[i])<(Battery_range[i][0]):
+      Reading_status[i]='L'
+      
+    elif (Battery_readings[i])>(Battery_range[i][1]):
+      Reading_status[i]='H'
```

Comment:
These lines are a 'conversion' from readings to `L` or `H`.
Separate that conversion function and you will control the complexity.
It also opens the possibility to test that function on its own, avoiding repetition of tests for every battery parameter.